api/src/main/java/com/openlattice/data/DataApi.java

```diff
@@ -195,12 +196,12 @@ Integer replaceAssociationData(
      * Deletes the entities matching the given entity ids and all of its neighbor entities provided in the filter.
      *
      * @param entitySetId The id of the EntitySet to delete from.
-     * @param filter EntityNeighboursFilter containing which ids of entities to delete and entity set ids of neighbours
-     * to delete from.
-     * @param deleteType The delete type to perform (soft or hard delete).
+     * @param filter      EntityNeighboursFilter containing which ids of entities to delete and entity set ids of neighbours
+     *                    to delete from.
+     * @param deleteType  The delete type to perform (soft or hard delete).
      */
-    @POST( BASE + "/" + ENTITY_SET + "/" + SET_ID_PATH + "/" + NEIGHBORS )
-    Long deleteEntitiesAndNeighbors(
+    @HTTP( method = "DELETE", path = BASE + "/" + ENTITY_SET + "/" + SET_ID_PATH + "/" + NEIGHBORS, hasBody = true )
+    UUID deleteEntitiesAndNeighbors(
```

Comment:
Need to update `openlattice.yaml`, specifically line 2149 in https://github.com/openlattice/openlattice/blob/6754c2a8899a8e619f09e24c4805836db23fe427/api/openlattice.yaml#L2148-L2151